app/models/sms/decoder/decoder.rb

```diff
@@ -125,7 +125,8 @@ def find_form
 
         # If form version outdated we must specify the form AND form_code in the error message
         # since they are different.
-        raise_decoding_error("form_version_outdated", form: v.form, form_code: code) unless v.is_current?
+        raise_decoding_error("form_version_outdated", form: v.form, form_code: code) unless
```

Comment:
Yikes, old one-letter var! Please rename `v` to `version` as a refactor.

Reply:
There is also a bunch of LoD violations in here. Can you assign `form = version.form` further up once we know we have a form, and then maybe add a delegator in form to create `minimum_version_number` since we use that in lots of places?